experimental/auth-logic/wrappers/provenance_wrapper.go

```diff
@@ -50,3 +57,19 @@ func (p provenanceWrapper) EmitStatement() (UnattributedStatement, error) {
 		Contents: fmt.Sprintf(`expected_hash("%s::Binary", sha256:%s).`, applicationName,
 			expectedHash)}, nil
 }
+
+// GetAppNameFromProvenance parses a provenance file and emits the name of the
+// application it is about. This is useful for generating principal names,
+// for example.
+func GetAppNameFromProvenance(provenanceFilePath string) (string, error) {
+	provenance, provenanceErr := slsa.ParseProvenanceFile(provenanceFilePath)
+	if provenanceErr != nil {
+		return "", provenanceErr
+	}
+	
+  if len(provenance.Subject) < 1 {
```

Comment:
The indentation does not look right, do the auto fmt tools not work with the Bazel structure?

Reply:
I just didn't run it yet, and figured you'd focus on other things. oops